core/io/remote_filesystem_client.cpp

```diff
@@ -0,0 +1,333 @@
+		file_cache_buffer.resize(res_len);
+
+		tcp_client->put_32(cs.length()); // Size of buffer uncompressed
+		tcp_client->put_32(file_cache_buffer.size()); // Size of buffer compressed
+		tcp_client->put_data(file_cache_buffer.ptr(), file_cache_buffer.size()); // Buffer
+	} else {
+		tcp_client->put_32(0); // No file cache buffer
+	}
+
+	tcp_client->poll();
+	ERR_FAIL_COND_V_MSG(tcp_client->get_status() != StreamPeerTCP::STATUS_CONNECTED, ERR_CONNECTION_ERROR, "Remote filesystem server disconnected after sending header.");
+
+	uint32_t file_count = tcp_client->get_32();
+
+	ERR_FAIL_COND_V_MSG(tcp_client->get_status() != StreamPeerTCP::STATUS_CONNECTED, ERR_CONNECTION_ERROR, "Remote filesystem server disconnected while waiting for file list");
+
+	LocalVector<uint8_t> file_buffer;
+
+	Vector<FileCache> temp_file_cache;
+
+	HashSet<String> files_processed;
+	for (uint32_t i = 0; i < file_count; i++) {
+		String file = tcp_client->get_utf8_string();
+		ERR_FAIL_COND_V_MSG(file == String(), ERR_CONNECTION_ERROR, "Invalid file name received from remote filesystem.");
+		uint64_t server_modified_time = tcp_client->get_u64();
+		ERR_FAIL_COND_V_MSG(tcp_client->get_status() != StreamPeerTCP::STATUS_CONNECTED, ERR_CONNECTION_ERROR, "Remote filesystem server disconnected while waiting for file info.");
+
+		FileCache fc;
+		fc.path = file;
+		fc.server_modified_time = server_modified_time;
+		temp_file_cache.push_back(fc);
+
+		files_processed.insert(file);
+	}
+
+	Vector<FileCache> new_file_cache;
+
+	// Get the actual files. As a robustness measure, if the connection is interrupted here, any file not yet received will be considered removed.
+	// Since the file changed anyway, this makes it the easiest way to keep robustness.
+
+	bool server_disconnected = false;
+	for (uint32_t i = 0; i < file_count; i++) {
+		String file = temp_file_cache[i].path;
+
+		if (temp_file_cache[i].server_modified_time == 0 || server_disconnected) {
+			// File was removed, or server disconnected before tranfering it. Since its no longer valid, remove anyway.
+			_remove_file(file);
+			continue;
+		}
+
+		uint64_t file_size = tcp_client->get_u64();
+		file_buffer.resize(file_size);
+		if (file_buffer.size() != file_size) {
+			server_disconnected = true;
+			ERR_PRINT("Remote filesystem sent a file (" + file + ") too big (Unable to allocate memory): " + String::humanize_size(file_size));
+		} else {
+			err = tcp_client->get_data(file_buffer.ptr(), file_size);
+			if (err != OK) {
+				ERR_PRINT("Error retrieving file from remote filesystem: " + file);
+				server_disconnected = true;
+			}
+		}
+
+		if (tcp_client->get_status() != StreamPeerTCP::STATUS_CONNECTED) {
+			// Early disconnect, stop accepting files.
+			server_disconnected = true;
+		}
+
+		if (server_disconnected) {
+			// No more server, transfer is invalid, remove this file.
+			_remove_file(file);
+			continue;
+		}
+
+		uint64_t modified_time;
```

Comment:
```suggestion
		uint64_t modified_time = 0;
```

Initializing eases debugging and it's lastely the trend in Godot.